profiles/apparmor/template.go

```diff
@@ -26,7 +26,7 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
 
   deny @{PROC}/* w,   # deny write for all files directly in /proc (not in a subdir)
   # deny write to files not in /proc/<number>/** or /proc/sys/**
-  deny @{PROC}/{[^1-9],[^1-9][^0-9],[^1-9s][^0-9y][^0-9s],[^1-9][^0-9][^0-9][^0-9]*}/** w,
+  deny @{PROC}/{[^1-9],[^1-9][^0-9],[^1-9s][^0-9y][^0-9s],[^1-9][^0-9][^0-9][^0-9/]*}/** w,
```

Comment:
I think it'd be a little bit easier to follow (though to be fair, not by much) if we added "sys" to the last case as well:

```
deny @{PROC}/{[^1-9],[^1-9][^0-9],[^1-9s][^0-9y][^0-9s],[^1-9s][^0-9y][^0-9s][^0-9/]*}/** w,
```
(AppArmor profiles make my eyes hurt :fire::eyes::fire:.)

Reply:
Ugh, um yeah thats a little clearer, maybe, I would like a comment explaining this

Reply:
Yeah maybe a paragraph. The fact this is necessary at all is completely insane...

Reply:
Maybe I'm failing to remember something, but was there a reason why we didn't split these into separate `deny` statements?

Reply:
IMO adding "sys" to the last alternative does not make it any easier to follow as it extends the do-not-deny list in odd directions: e.g. `/proc/sys3foo/bar` would be not-denied by such a rule. What I think would help make it easier to understand would be to restrict all the patterns to match within a single path component. Exploding the alternation into separate rules might also help, especially as it gives more room for comments.
```suggestion
  deny @{PROC}/[^1-9]/** w,  # deny /proc/?/** except /proc/<pid>/**
  deny @{PROC}/[^1-9][^0-9/]/** w,  # deny /proc/??/** except /proc/<pid>/**
  deny @{PROC}/[^1-9s][^0-9y/][^0-9s/]/** w,  # deny /proc/???/** except /proc/<pid>/** or /proc/sys/**
  deny @{PROC}/[^1-9][^0-9/][^0-9/][^0-9/]*/** w,  # deny /proc/???*/** except /proc/<pid>/**
```